src/game/server/entities/character.cpp

```diff
@@ -1257,7 +1257,7 @@ void CCharacter::Snap(int SnappingClient)
 		pDDNetCharacter->m_Flags |= CHARACTERFLAG_NO_HOOK;
 	if(m_SuperJump)
 		pDDNetCharacter->m_Flags |= CHARACTERFLAG_ENDLESS_JUMP;
-	if(m_Jetpack || m_NinjaJetpack)
+	if(m_Jetpack || (m_NinjaJetpack && !m_Core.m_HasTelegunGun))
```

Comment:
This looks incorrect, and was also incorrect before.

Whether the player enabled ninja jetpack or not should have no influence over whether the jetpack state is sent to the client.

```suggestion
	if(m_Jetpack)
```